### DISCUSSION TITLE
Verwaltung von erforderlichen Python-Paketen mit „requirements.txt“

### DISCUSSION THREAD

**Discussion by pjauch** [i4s-htwk/sStatics#3] [Wiki]
Die Datei `requirements.txt` dient in Python-Projekten dazu, alle externen Abhängigkeiten (Pakete) zu dokumentieren, die zur Ausführung des Projekts erforderlich sind. Dies erleichtert die Installation der benötigten Pakete in anderen Umgebungen oder durch andere Nutzer des Projekts.
## Erstellen einer `requirements.txt`

Um die in der aktuellen virtuellen Umgebung installierten Pakete in der Datei `requirements.txt` zu speichern, kann folgender Befehl verwendet werden:
```shell
pip freeze > requirements.txt
```
Dieser Befehl listet alle derzeit installierten Pakete sowie deren genaue Versionsnummern auf und speichert sie in der Datei.

## Aktualisierung der requirements.txt
Jedes Mal, wenn ein neues Paket zur Projektumgebung hinzugefügt wird, sollte die `requirements.txt` aktualisiert werden, damit alle Abhängigkeiten konsistent bleiben. Dazu wird erneut der Befehl `pip freeze > requirements.txt` ausgeführt. Ein vorheriges Löschen der bestehenden Datei ist nicht notwendig, da der Befehl die Datei automatisch überschreibt.

**Comment by pbrassel** [comment]
Mittels `requirements.txt` ist es möglich, eine virtuelle Umgebung 1:1 zu spiegeln. Mit Verweis auf [die beiden unterschiedlichen Setups](https://github.com/sStatics/sstatics-intern/discussions/2) (Nutzer- und Entwicklerinstallation), kann `requirements.txt` während der Entwicklerinstallation zur Installation aller Pakete durch `pip install -r requirements.txt` verwendet werden. Für den normalen Nutzer spielt `requirements.txt` **keine** Rolle.
